app/src/main/java/com/github/se/project/ui/overview/Overview.kt

```diff
@@ -0,0 +1,165 @@
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.lazy.LazyColumn
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.filled.AccountBox
+import androidx.compose.material3.Button
+import androidx.compose.material3.Card
+import androidx.compose.material3.Icon
+import androidx.compose.material3.IconButton
+import androidx.compose.material3.Scaffold
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.collectAsState
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.unit.dp
+import com.github.se.project.model.lesson.Lesson
+import com.github.se.project.model.lesson.LessonViewModel
+import com.github.se.project.model.profile.ListProfilesViewModel
+import com.github.se.project.model.profile.Profile
+import com.github.se.project.model.profile.Role
+import com.github.se.project.ui.navigation.BottomNavigationMenu
+import com.github.se.project.ui.navigation.LIST_TOP_LEVEL_DESTINATIONS_STUDENT
+import com.github.se.project.ui.navigation.LIST_TOP_LEVEL_DESTINATIONS_TUTOR
+import com.github.se.project.ui.navigation.NavigationActions
+import com.github.se.project.ui.navigation.Screen
+
+@Composable
+fun HomeScreen(
+    listProfilViewModele: ListProfilesViewModel,
+    lessonViewModel: LessonViewModel,
+    navigationActions: NavigationActions,
+) {
+
+  val context = LocalContext.current
+  val currentProfile = listProfilViewModele.currentProfile.collectAsState().value
+  val lessons = lessonViewModel.currentUserLessons.collectAsState().value
+
+  // lessonViewModel.getLessonsByUserId(currentProfile?.uid ?: "")
+  when (currentProfile?.role) {
+    Role.TUTOR -> lessonViewModel.getLessonsForTutor(currentProfile.uid)
+    Role.STUDENT -> lessonViewModel.getLessonsForStudent(currentProfile.uid)
+    Role.UNKNOWN -> Toast.makeText(context, "Unknown Profil", Toast.LENGTH_SHORT).show()
+    null -> Toast.makeText(context, "null Profil", Toast.LENGTH_SHORT).show()
+  }
+
+  // Determine which bottom navigation items to show based on the user's role
+  val LIST_TOP_LEVEL_DESTINATIONS =
+      when (currentProfile?.role) {
+        Role.TUTOR -> LIST_TOP_LEVEL_DESTINATIONS_TUTOR
+        Role.STUDENT -> LIST_TOP_LEVEL_DESTINATIONS_STUDENT
+        Role.UNKNOWN -> LIST_TOP_LEVEL_DESTINATIONS_STUDENT
+        null -> LIST_TOP_LEVEL_DESTINATIONS_STUDENT
+      }
+
+  Scaffold(
+      topBar = {
+        Row(
+            modifier = Modifier.fillMaxWidth().padding(16.dp),
+            horizontalArrangement = Arrangement.End,
+            verticalAlignment = Alignment.CenterVertically) {
+              IconButton(
+                  onClick = {
+                    Toast.makeText(context, "Profile icon clicked", Toast.LENGTH_SHORT).show()
```

Comment:
now that the navigation is implemented, we might want to remove that toast :)

Reply:
Good point